components/tarotSpreadSelectionBox/TarotSpreadSelectionBox.tsx

```diff
@@ -1,224 +1,123 @@
   onClick: () => void;
 }
-interface TarotSpreadSpreadSelectionProps extends TarotSpreadAction {
+
+export interface TarotSpreadSelectionProps extends TarotSpreadAction {
   title: string;
   amountOfCards?: number;
   backgroundColor?: string;
+  zIndex?: number;
   description: string;
+  onClick: () => void;
 }
 
-const TarotSpreadSelectionBox = ({
-  amountOfCards,
-  title,
+const TarotSpreadSelectionBox: React.FC<TarotSpreadSelectionProps> = ({
   backgroundColor,
+  amountOfCards,
   description,
-  onClick
-}: TarotSpreadSpreadSelectionProps) => {
-  const tarotDeckData = useContext(TarotDeckContext);
+  onClick,
+  zIndex,
+  title
+}) => {
+  const tarotDeckData: TarotDeckData = useContext(TarotDeckContext);
+  const [isHovered, setIsHovered] = useState<boolean>(false);
   const { palette } = useTheme();
-  const [isHovered, setIsHovered] = useState(false);
+  const cardTransitionTime = '250ms';
+  const gridTransitionTime = '400ms';
+  const layout = getLayoutStyles(
+    tarotDeckData,
+    amountOfCards,
+    cardTransitionTime,
+    isHovered
+  );
+
   return (
     <Box
       onClick={onClick}
-      sx={{ width: 'fit-content' }}
-      onMouseEnter={() => {
-        setIsHovered(true);
-      }}
-      onMouseLeave={() => {
-        setIsHovered(false);
+      sx={{
+        width: '340px',
+        backgroundColor: palette.secondary.main,
+        borderRadius: '20px',
+        cursor: 'pointer',
+        transition: `
+          box-shadow ${gridTransitionTime},
+          transform ${gridTransitionTime}
+        `,
+        boxShadow: isHovered ? '3' : '1',
+        transform: isHovered ? 'scale(1.025) ' : '',
+        zIndex: zIndex
       }}
+      onMouseEnter={() => setIsHovered(true)}
+      onMouseLeave={() => setIsHovered(false)}
     >
       <Box
-        gap='8px'
-        flexDirection='column'
-        justifyContent='center'
-        position='relative'
         sx={{
+          padding: '60px 0',
+          borderTopLeftRadius: '20px',
+          borderTopRightRadius: '20px',
           backgroundColor: { backgroundColor },
           backgroundImage:
-            'linear-gradient(to right bottom, #f2afda, #ffb3c2, #ffbeb0, #f7cba9, #e7d8ae)',
-          borderTopLeftRadius: '10%',
-          borderTopRightRadius: '10%'
+            'linear-gradient(to right bottom, #f2afda, #ffb3c2, #ffbeb0, #f7cba9, #e7d8ae)'
         }}
       >
-        <Typography
-          pt='20px'
-          fontSize={30}
-          color='grey'
-          variant='h6'
-          textAlign='center'
-        ></Typography>
         <Box
-          position='relative'
-          width={325}
-          height={300}
-          display='flex'
-          alignItems='center'
+          display='grid'
           justifyContent='center'
+          gridTemplateColumns={
+            isHovered && layout.gridHover
+              ? `${layout.gridHover} 0fr`
+              : `${layout.gridTemplateColumns || 'auto'} 0fr`
+          }
+          sx={{
```

Comment:
Maybe this is a preference thing but if we have some styles should we put everything inside of the SX?